File: Java/ListingToDosJava/src/main/java/com/listingtodosjava/listingtodosjava/Controllers/ToDoController.java
package com.listingtodosjava.listingtodosjava.Controllers;

import com.listingtodosjava.listingtodosjava.Models.ToDos;
import com.listingtodosjava.listingtodosjava.Repositories.ToDoRepository;
import com.listingtodosjava.listingtodosjava.Services.AssigneeService;
import com.listingtodosjava.listingtodosjava.Services.ToDoService;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.servlet.ModelAndView;

@Controller
@AllArgsConstructor
@RequestMapping(path = "/")
public class ToDoController {

    private ToDoService toDoService;
    private AssigneeService assigneeService;
    private ToDoRepository toDoRepository;

    @GetMapping()
    public ModelAndView ListToDos(){

        ModelAndView mav = new ModelAndView("ToDo/ToDoList");

        mav.addObject("todos", toDoRepository.findAll());

        return mav;
    }

    @GetMapping("/ListUndone")
    public String ListUndone(){

        return "ToDoList";
    }

    @GetMapping("/AddToDo")
    public String Add(){

        return "AddToDo";
    }

    @PostMapping("/AddedToDo")
    public String AddToDo(String title, String description, Boolean urgent, Boolean done, Long assigneeId){

        return "ToDoList";
    }

    @GetMapping("/{id}/remove")
    public String RemoveToDo(Long id){

        return "ToDoList";
    }

    @GetMapping("/{id}/edit")
    public String EditToDo(Long id){

        return "EditToDo";
    }

    @PostMapping("/{id}/edited")
    public String EditedToDo(ToDos toDo, Long assigneeId){

        return "ToDoList";
    }

    @PostMapping("/search")
    public String Search(String searchInput){

        return "ToDoList";
    }

}

File: C#/OrientationPhasePractice/ListingToDos/ListingToDos/Controllers/ToDoController.cs
using ListingToDos.Models;
using ListingToDos.Services;
using Microsoft.AspNetCore.Mvc;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ListingToDos.Controllers
{
    [Route("/")]
    public class ToDoController : Controller
    {
        private ToDoService ToDoService { get; }
        private AssigneeService AssigneeService { get; }
        public ToDoController(ToDoService toDoService, AssigneeService assigneeService)
        {
            ToDoService = toDoService;
            AssigneeService = assigneeService;
        }
        public IActionResult List()
        {
            var todos = ToDoService.ListToDos();
            var assignees = AssigneeService.Find();
            ViewModel viewModel = new();
            viewModel.AllToDos = todos;
            viewModel.AllAssignees = assignees;
            return View(viewModel);
        }
        [HttpGet("/ListUndone")]
        public IActionResult ListUndone()
        {
            var todos = ToDoService.ListUndone();
            var assignees = AssigneeService.Find();
            ViewModel viewModel = new();
            viewModel.AllToDos = todos;
            viewModel.AllAssignees = assignees;
            return View("List", viewModel);
        }
        [HttpGet("/Add")]
        public IActionResult Add()
        {
            var assignees = AssigneeService.Find();
            ViewModel viewModel = new();
            viewModel.AllAssignees = assignees;
            return View(viewModel);
        }
        [HttpPost("/AddToDo")]
        public IActionResult AddToDo(string title, string description, bool isUrgent, long assigneeKey)
        {
            ToDoService.AddToDo(title, description, isUrgent, assigneeKey);
            return RedirectToAction("List");
        }
        [HttpGet("/{id}/remove")]
        public IActionResult RemoveToDo([FromRoute]long id)
        {
            ToDoService.RemoveToDo(id);
            return RedirectToAction("List");
        }
        [HttpGet("/{id}/edit")]
        public IActionResult EditToDo([FromRoute]long id)
        {
            var todo = ToDoService.Find(id);
            var assignees = AssigneeService.Find();
            var assignee = AssigneeService.FindSpec(todo.AssigneeId);
            ViewModel viewModel = new();
            viewModel.ToDo = todo;
            viewModel.AllAssignees = assignees;
            viewModel.Assignee = assignee;
            return View(viewModel);
        }
        [HttpPost("/{id}/edited")]
        public IActionResult EditedToDo(ToDo todo, long assigneeKey)
        {
            ToDoService.Edit(todo, assigneeKey);
            return RedirectToAction("List");
        }
        [HttpPost("/search")]
        public IActionResult Search(string searchInput)
        {
            var todos = ToDoService.Search(searchInput);
            var assignees = AssigneeService.Find();
            ViewModel viewModel = new();
            viewModel.AllToDos = todos;
            viewModel.AllAssignees = assignees;
            return View(viewModel);
        }
    }
}
